openfl/utilities/verifdataset/verifiable_dataset.py

```diff
@@ -0,0 +1,164 @@
+# Copyright 2020-2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+""" This module contains the VerifiableDataset class. """
+
+import json
+from enum import Enum
+from hashlib import sha384
+from pathlib import Path
+from typing import List
+
+from openfl.utilities.verifdataset.data_source import DataSource, DataSourceType
+from openfl.utilities.verifdataset.local_data_source import LocalDataSource
+
+
+class DatasetFormat(Enum):
+    """Enum for the different dataset commitment formats."""
+
+    VERBOSE = "verbose_dataset"
+    CONCISE = "concise_dataset"
+
+
+def filter_non_serializable(obj):
+    """Filter out methods and non-serializable objects."""
+    serializable_dict = {}
+    for key, val in obj.__dict__.items():
+        if callable(val):
+            continue  # Skip methods
+        if isinstance(val, Enum):
+            val = val.value  # Convert Enum to its value
+        elif isinstance(val, Path):
+            val = str(val)  # Convert Path to string
+        serializable_dict[key] = val
+    return serializable_dict
+
+
+class VerifiableDataset:
```

Comment:
@Efrat1 and @vrancurel, if we wanted to introduce Merkle tree verification, I suppose we could extend this class to a `MerkleVerifiableDataset` and override the `verify_dataser` and `verify_single_file` methods? What other changes would be involved?

Reply:
We need to define an abstract interface "commitment" with 2 methods create and verify. We need to think this through. For now we can merge as is, as the focus will be S3. We can create the commitment interface after.

Reply:
Sounds good, thanks for the clarification!